tests/bsim/bluetooth/mesh/src/test_advertiser.c

```diff
@@ -453,18 +453,19 @@ static void test_rx_proxy_mixin(void)
 	/* Delay to provision dut */
 	k_sleep(K_MSEC(1000));
 
+	/* Scan adv data. */
+	xmit_param.retr = 5;
+	xmit_param.interval = 20;
+
+	ASSERT_OK(bt_mesh_test_wait_for_packet(xmit_scan_cb, &observer_sem, 20));
+
 	/* Scan proxy beacons. */
 	/* (total transmit duration) / (transmit interval) */
 	gatt_param.transmits = 5000 / 1000;
 	gatt_param.interval = 1000;
 	gatt_param.service = MESH_SERVICE_PROXY;
 	ASSERT_OK(bt_mesh_test_wait_for_packet(gatt_scan_cb, &observer_sem, 20));
 
-	/* Scan adv data. */
-	xmit_param.retr = 5;
-	xmit_param.interval = 20;
-	ASSERT_OK(bt_mesh_test_wait_for_packet(xmit_scan_cb, &observer_sem, 20));
-
 	/* Scan proxy beacons again. */
```

Comment:
I checked what happened in the test:
```
d_00: @00:00:01.828003  [00:00:01.828,002] <err> bt_mesh_adv_ext: Failed updating adv params: -105
d_00: @00:00:01.828003  [00:00:01.828,002] <wrn> bt_mesh_gatt: Failed to advertise using Network ID (err -105)
d_00: @00:00:01.828003  [00:00:01.828,002] <err> bt_mesh_gatt: Advertising proxy failed (err: -105)
```

We need to increase TX CMD buffers. Adding this to `overlay_gatt.conf` fixes the problem and then this change is not needed:
```
diff --git a/tests/bsim/bluetooth/mesh/overlay_gatt.conf b/tests/bsim/bluetooth/mesh/overlay_gatt.conf
index a8ce71afaef..d6c23ac1a0a 100644
--- a/tests/bsim/bluetooth/mesh/overlay_gatt.conf
+++ b/tests/bsim/bluetooth/mesh/overlay_gatt.conf
@@ -5,3 +5,5 @@ CONFIG_BT_MESH_PB_GATT=y
 CONFIG_BT_CENTRAL=y
 CONFIG_BT_MESH_PROXY_CLIENT=y
 CONFIG_BT_MESH_PROXY_SOLICITATION=y
+
+CONFIG_BT_BUF_CMD_TX_COUNT=3
```

Reply:
Could you elaborate a bit on how those errors and warnings point to TX CMD buffers being too small? Wouldn't get that from the logs.

Reply:
The failed variant runs the mesh advertiser on the sysworkq:
https://github.com/zephyrproject-rtos/zephyr/blob/f5305717c44e5a4aab619cbaf779c220ccc1c8a5/tests/bsim/bluetooth/mesh/tests_scripts/advertiser/proxy_mixin.sh#L24-L25

Which is yet another outcome of this: https://github.com/zephyrproject-rtos/zephyr/issues/77241

This time the stack competes with itself. The start of the scanner "eats" all buffers, then goes to sleep, which wakes up the advertiser thread. And the advertiser doesn't have enough buffers to update adv params.

Backtrace of scanner start:
```
#12 0x0804d7df in k_sem_take (sem=0xf57fefa4, timeout=...) at zephyr/include/generated/zephyr/syscalls/kernel.h:1086                                                                                        
#13 0x08097986 in bt_hci_cmd_send_sync (opcode=8257, buf=0x838a3c0 <_net_buf_hci_cmd_pool>, rsp=0x0) at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/host/hci_core.c:434          
#14 0x080ae733 in start_le_scan_ext (scan_param=0xf57ff080) at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/host/scan.c:269                                                       
#15 0x080aea9d in start_scan (scan_param=0xf57ff080) at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/host/scan.c:388                                                              
#16 0x080af012 in scan_update () at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/host/scan.c:465                                                                                  
#17 0x080af362 in bt_le_scan_user_add (flag=BT_LE_SCAN_USER_EXPLICIT_SCAN) at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/host/scan.c:512                                        
#18 0x080b1c7c in bt_le_scan_start (param=0xf57ff120, cb=0x80eae97 <bt_mesh_scan_cb>) at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/host/scan.c:1770                            
#19 0x080eb4d0 in bt_mesh_scan_enable () at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/adv.c:395                                                                           
#20 0x08133b6c in bt_mesh_lpn_init () at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/lpn.c:1184                                                                             
#21 0x080e8d0f in bt_mesh_start () at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/main.c:653                                                                                
#22 0x080e6c72 in bt_mesh_provision (net_key=0x827606c <test_net_key> "\001\002\003", net_idx=0, flags=0 '\000', iv_index=0, addr=1, dev_key=0x827608e <adv_cfg+2> "\001") at /home/pavel/nordic/ncs/git-rep
os/zephyrproject/zephyr/subsys/bluetooth/mesh/main.c:166
```

Backtrace of failed adv params update:
```
#0  adv_start (ext_adv=0x82c1d60 <advs>, param=0xf61fe0f0, start=0xf61fe098, ad=0x827ae3c <net_id_ad>, ad_len=3, sd=0x0, sd_len=0) at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth
/mesh/adv_ext.c:177                                                                                                                                                                                         
#1  0x0812d3f3 in bt_mesh_adv_gatt_start (param=0xf61fe0f0, duration=-1, ad=0x827ae3c <net_id_ad>, ad_len=3, sd=0x0, sd_len=0) at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mes
h/adv_ext.c:581                                                                                                                                                                                             
#2  0x08154921 in net_id_adv (sub=0x82bf1a0 <subnets>, duration=-1) at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/proxy_srv.c:628
#3  0x08155a7f in gatt_proxy_advertise () at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/proxy_srv.c:865
#4  0x0815700c in bt_mesh_proxy_adv_start () at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/proxy_srv.c:1206
#5  0x080eadb3 in bt_mesh_adv_gatt_send () at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/adv.c:302
#6  0x0812c378 in send_pending_adv (work=0x82c1d74 <advs+20>) at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/adv_ext.c:351
#7  0x08203fe6 in work_queue_main (workq_ptr=0x836c800 <k_sys_work_q>, p2=0x0, p3=0x0) at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/kernel/work.c:684
```